hazelcast/src/main/java/com/hazelcast/map/impl/SizeEstimator.java

```diff
@@ -23,11 +23,17 @@
  */
 public interface SizeEstimator<T> {
```

Comment:
The name of this class is not self documenting for the following reasons:

* it doesn't track size, it tracks memory usage. Size in map and record store is being used to define the number of elements; not the memory usage. So the name 'size' is misleading.
* Size estimator gives impression of being a function like objects, so not tracking any state. However in this case you are tracking size state to prevent doing a full scan (which is good). However the name of the class doesn't reveal this. 

I would rename it to StorageMemoryUsageTracker.

Using this name it automatically resolves the issues above. Also is makes it immediately clear that it tracks it not from a map, but from a Storage. Afaik Storage is only used by single thread at any given moment, so therefor doesn't need to be threadsafe.